File: examples/test_strings_runner.cpp
#include "test_strings.hpp"
#include <iostream>
#include <vector>
#include <cassert>

using namespace teststrings;

int main() {
    std::cout << "=== Testing String Types ===\n\n";

    // Create test data
    FileHeader original;
    original.signature = "DEZZ";
    original.name_len = 8;
    original.filename = "test.dat";
    original.path = "/usr/local/bin";

    // Write to bytes
    Writer writer;
    original.write(writer);
    std::vector<uint8_t> data = writer.finish();

    std::cout << "Wrote " << data.size() << " bytes\n";
    std::cout << "  signature: \"" << original.signature << "\"\n";
    std::cout << "  name_len: " << (int)original.name_len << "\n";
    std::cout << "  filename: \"" << original.filename << "\"\n";
    std::cout << "  path: \"" << original.path << "\"\n\n";

    // Read it back
    Reader reader(data);
    FileHeader parsed = FileHeader::read(reader);

    std::cout << "Read back:\n";
    std::cout << "  signature: \"" << parsed.signature << "\"\n";
    std::cout << "  name_len: " << (int)parsed.name_len << "\n";
    std::cout << "  filename: \"" << parsed.filename << "\"\n";
    std::cout << "  path: \"" << parsed.path << "\"\n\n";

    // Verify
    assert(parsed.signature == original.signature);
    assert(parsed.name_len == original.name_len);
    assert(parsed.filename == original.filename);
    assert(parsed.path == original.path);

    std::cout << "=== All string tests passed! ===\n";
    return 0;
}

File: examples/test_strings_runner.py
#!/usr/bin/env python3
"""Test runner for string types"""

from test_strings import FileHeader

def main():
    print("=== Testing String Types ===\n")

    # Create test data
    original = FileHeader(
        signature="DEZZ",
        name_len=8,
        filename="test.dat",
        path="/usr/local/bin"
    )

    # Write to bytes
    data = original.write()

    print(f"Wrote {len(data)} bytes")
    print(f"  signature: \"{original.signature}\"")
    print(f"  name_len: {original.name_len}")
    print(f"  filename: \"{original.filename}\"")
    print(f"  path: \"{original.path}\"\n")

    # Read it back
    parsed, bytes_read = FileHeader.read(data)

    print("Read back:")
    print(f"  signature: \"{parsed.signature}\"")
    print(f"  name_len: {parsed.name_len}")
    print(f"  filename: \"{parsed.filename}\"")
    print(f"  path: \"{parsed.path}\"\n")

    # Verify
    assert parsed.signature == original.signature, "Signature mismatch"
    assert parsed.name_len == original.name_len, "Name length mismatch"
    assert parsed.filename == original.filename, "Filename mismatch"
    assert parsed.path == original.path, "Path mismatch"

    print("=== All string tests passed! ===")

if __name__ == "__main__":
    main()
